src/parallel/HighsTaskExecutor.h

```diff
@@ -85,47 +90,75 @@ class HighsTaskExecutor {
-    // Following yields warning C4706: assignment within conditional
-    // expression when building libhighs on Windows (/W4):
-    //
-    //    while (!(executor = mainWorkerHandle.load(std::memory_order_acquire)))
-    //      HighsSpinMutex::yieldProcessor();
-    while (true) {
-      executor = mainWorkerHandle.load(std::memory_order_acquire);
-      if (executor != nullptr) break;
-      HighsSpinMutex::yieldProcessor();
-    }
-    // now acquire a reference count of the global executor
-    threadLocalExecutorHandle() = *executor;
-    HighsSplitDeque* localDeque = workerDeques[workerId].get();
-    threadLocalWorkerDeque() = localDeque;
-    HighsTask* currentTask = workerBunk->waitForNewTask(localDeque);
-    while (currentTask != nullptr) {
-      localDeque->runStolenTask(currentTask);
-
-      currentTask = random_steal_loop(localDeque);
-      if (currentTask != nullptr) continue;
-
-      currentTask = workerBunk->waitForNewTask(localDeque);
+  static void run_worker(int workerId, HighsTaskExecutor* ptr) {
+    auto& executorHandle = threadLocalExecutorHandle();
+    executorHandle.ptr = ptr;
+
+    // check if main thread has shutdown before thread has started
+    // if (ptr->mainWorkerId.load() != std::thread::id()) {
+    if (!(executorHandle.isMain)) {
+      HighsSplitDeque* localDeque = ptr->workerDeques[workerId].get();
+      threadLocalWorkerDeque() = localDeque;
+
+      HighsTask* currentTask = ptr->workerBunk->waitForNewTask(localDeque);
+      while (currentTask != nullptr) {
+        localDeque->runStolenTask(currentTask);
+
+        currentTask = ptr->random_steal_loop(localDeque);
+        if (currentTask != nullptr) continue;
+
+        currentTask = ptr->workerBunk->waitForNewTask(localDeque);
+      }
     }
+
+    threadLocalExecutorHandle().dispose();
   }
 
  public:
   HighsTaskExecutor(int numThreads) {
     assert(numThreads > 0);
-    mainWorkerHandle.store(nullptr, std::memory_order_relaxed);
+
     workerDeques.resize(numThreads);
     workerBunk = cache_aligned::make_shared<HighsSplitDeque::WorkerBunk>();
     for (int i = 0; i < numThreads; ++i)
       workerDeques[i] = cache_aligned::make_unique<HighsSplitDeque>(
           workerBunk, workerDeques.data(), i, numThreads);
 
     threadLocalWorkerDeque() = workerDeques[0].get();
-    for (int i = 1; i < numThreads; ++i)
-      std::thread([&](int id) { run_worker(id); }, i).detach();
+    workerThreads.reserve(numThreads - 1);
+    referenceCount.store(numThreads);
+
+    for (int i = 1, numThreads = workerDeques.size(); i < numThreads; ++i) {
+      workerThreads.emplace_back(
+          std::move(std::thread(&HighsTaskExecutor::run_worker, i, this)));
+    }
+  }
+
+  void stopWorkerThreads(bool blocking = false) {
+    // auto id = mainWorkerId.exchange(std::thread::id());
+    // if (id == std::thread::id()) return;  // already been called
+    auto& executorHandle = threadLocalExecutorHandle();
+    if (executorHandle.isStopped) return;
```

Comment:
Need to "safely" only call stop once. i.e., 
```cpp
if (executorHandle.ptr == nullptr || executorHandle.ptr->isStopped.exchange(true)) 
   return;
```

This will set `isStopped` to `true` and return the previous result.  If it had already been set to `true` we will exit, otherwise we'll proceed to shut things down.  This should be guaranteed to avoid race conditions and only shut down once.